docs/reference.rst

```diff
@@ -0,0 +1,113 @@
+.. _sec_ref:
+
+Reference
+=========
+
+.. contents::
+   :local:
+   :depth: 1
+
+.. _sec_ref_config_file:
+
+Configuration File
+------------------
+
+Example configuration:
+
+.. code-block:: cfg
+
+  [client]
+  hawkbit_server            = 127.0.0.1:8080
+  ssl                       = false
+  ssl_verify                = false
+  tenant_id                 = DEFAULT
+  target_name               = test-target
+  auth_token                = bhVahL1Il1shie2aj2poojeChee6ahShu
+  #gateway_token            = bhVahL1Il1shie2aj2poojeChee6ahShu
+  bundle_download_location  = /tmp/bundle.raucb
+  retry_wait                = 60
+  connect_timeout           = 20
+  timeout                   = 60
+  log_level                 = debug
+
+  [device]
+  product                   = Terminator
+  model                     = T-1000
+  serialnumber              = 8922673153
+  hw_revision               = 2
+  key1                      = value
+  key2                      = value
+
+**[client] section**
+
+Configures how to connect with the hawkBit server, etc.
+
+Mandatory options:
+
+``hawkbit_server=<host>[:<port>]``
+  The ip or hostname of the hawkbitServer to connect to.
+  The ``port`` can be provided optionally.
+  It needs to be separated with colons.
+
+``target_name=<name>``
+  Unique ``name`` string to identify controller.
+
+``auth_token=<token>``
+  Controller-specific authentication token.
+  This is set for each device individually.
+
+  .. note:: Either ``auth_token`` or ``gateway_token`` must be provided
+
+``gateway_token``
+  Gateway authentication token.
+  This is set by hawkBit.
```

Comment:
What does this mean? This can be set just like the authentication token on hawkBit side (via the [managament API](https://www.eclipse.org/hawkbit/apis/management_api/)) or generated randomly by hawkBit. Maybe better mention that this is a tenant-wide token and it must be enabled first?

Reply:
You're right. I've tried to be more specific here and also linked the hawkbit doc for authentication.